GH checks on the PR failing

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

Standalone DSPO (without ODH)

### Version

V1.5.0

### Environment

GH Actions/workflows.

### Current Behavior

The GH action checks on PRs to this repository are failing with Unit Tests and Functional Tests with the following errors:
Error: ../../../go/pkg/mod/go.uber.org/multierr@v1.10.0/error.go:224:20: undefined: atomic.Bool
note: module requires Go 1.19
make: *** [Makefile:198: /home/runner/work/data-science-pipelines-operator/data-science-pipelines-operator/bin/setup-envtest] Error 2
Error: Process completed with exit code 2.

### Expected Behavior

All the CI checks should pass.

### Steps To Reproduce

Create a new PR for this repo. 

### Workaround (if any)

_No response_

### Anything else

Following are the PRs which has the mentioned issue:

https://github.com/opendatahub-io/data-science-pipelines-operator/pull/394

https://github.com/opendatahub-io/data-science-pipelines-operator/pull/391

--- a/.github/workflows/functests.yml
+++ b/.github/workflows/functests.yml
@@ -9,6 +9,6 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: '1.18.x'
+          go-version: '1.19.x'
       - name: Run Functional Tests
         run: make functest

--- a/.github/workflows/precommit.yml
+++ b/.github/workflows/precommit.yml
@@ -5,7 +5,7 @@ jobs:
   precommit:
     runs-on: ubuntu-latest
     container:
-      image: quay.io/opendatahub/pre-commit-go-toolchain:v0.2
+      image: quay.io/opendatahub/pre-commit-go-toolchain:v0.3
       env:
         XDG_CACHE_HOME: /cache
         GOCACHE: /cache/go-build

--- a/.github/workflows/precommit.yml
+++ b/.github/workflows/precommit.yml
@@ -20,5 +20,7 @@ jobs:
         with:
           path: /cache
           key: ${{ runner.os }}-cache-${{ hashFiles('**/go.sum', '.pre-commit-config.yaml') }}
+      - name: Mark source directory as safe
+        run: git config --global --add safe.directory $GITHUB_WORKSPACE
       - name: Run pre-commit checks
         run: pre-commit run --all-files

--- a/.github/workflows/unittests.yml
+++ b/.github/workflows/unittests.yml
@@ -9,6 +9,6 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: '1.18.x'
+          go-version: '1.19.x'
       - name: Run Unit Tests
         run: make unittest

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.18.9-8 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.19.10 as builder
 ARG TARGETOS
 ARG TARGETARCH
 

--- a/controllers/testutil/util.go
+++ b/controllers/testutil/util.go
@@ -20,7 +20,6 @@ import (
 	"context"
 	"fmt"
 
-	"io/ioutil"
 	"os"
 	"time"
 
@@ -167,14 +166,14 @@ func DirExists(path string) (bool, error) {
 func GenerateDeclarativeTestCases() []Case {
 	var testcases []Case
 
-	cases, err := ioutil.ReadDir(CasesDir)
+	cases, err := os.ReadDir(CasesDir)
 	Expect(err).ToNot(HaveOccurred(), "Failed to fetch cases in case dir.")
 	for _, testcase := range cases {
 		caseName := testcase.Name()
 		caseDir := fmt.Sprintf("%s/%s", CasesDir, caseName)
 		newCase := Case{}
 		caseDeployDir := fmt.Sprintf("%s/deploy", caseDir)
-		deploys, err := ioutil.ReadDir(caseDeployDir)
+		deploys, err := os.ReadDir(caseDeployDir)
 		Expect(err).ToNot(HaveOccurred(), "Failed to read case.")
 		for _, f := range deploys {
 			newCase.Deploy = append(newCase.Deploy, fmt.Sprintf("%s/%s", caseDeployDir, f.Name()))
@@ -184,7 +183,7 @@ func GenerateDeclarativeTestCases() []Case {
 		caseCreationsFound, err := DirExists(caseCreateDir)
 		Expect(err).ToNot(HaveOccurred(), "Failed to read 'create' dir.")
 		if caseCreationsFound {
-			toCreate, err := ioutil.ReadDir(caseCreateDir)
+			toCreate, err := os.ReadDir(caseCreateDir)
 			Expect(err).ToNot(HaveOccurred(), "Failed to read 'create' dir.")
 			for _, f := range toCreate {
 				newCase.Expected.Created = append(newCase.Expected.Created, fmt.Sprintf("%s/%s", caseCreateDir, f.Name()))
@@ -195,7 +194,7 @@ func GenerateDeclarativeTestCases() []Case {
 		caseNoCreationsFound, err := DirExists(caseNotCreateDir)
 		Expect(err).ToNot(HaveOccurred(), "Failed to read 'not_create' dir.")
 		if caseNoCreationsFound {
-			toNotCreate, err := ioutil.ReadDir(caseNotCreateDir)
+			toNotCreate, err := os.ReadDir(caseNotCreateDir)
 			Expect(err).ToNot(HaveOccurred(), "Failed to read 'not_create' dir.")
 			for _, f := range toNotCreate {
 				newCase.Expected.NotCreated = append(newCase.Expected.NotCreated, fmt.Sprintf("%s/%s", caseNotCreateDir, f.Name()))

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/opendatahub-io/data-science-pipelines-operator
 
-go 1.18
+go 1.19
 
 require (
 	github.com/fsnotify/fsnotify v1.5.4

